Use simpler initialization of std::set

kronbichler opened the pull request:
As suggested by https://github.com/dealii/dealii/pull/13674#issuecomment-1118646185 - thanks @bangerth for the good hint for modernizing things.

marcfehling commented on tests/matrix_free/copy_feevaluation.cc:
```diff
@@ -172,8 +172,7 @@ test()
 
-  std::set<types::boundary_id> no_normal_flux_boundaries;
-  no_normal_flux_boundaries.insert(1);
+  std::set<types::boundary_id> no_normal_flux_boundaries = {1};
```
```suggestion
  const std::set<types::boundary_id> no_normal_flux_boundaries = {1};
```
I think `const` should work here, too.

kronbichler replied:
I agree. I widened my search and found some more appearances.

marcfehling commented:
```
/jenkins/workspace/dealii_PR-13676/tests/dofs/dof_tools_09.cc: In function ‘void check_this(const dealii::DoFHandler<dim>&)’:
/jenkins/workspace/dealii_PR-13676/tests/dofs/dof_tools_09.cc:45:24: error: no matching function for call to ‘std::set<unsigned int>::insert(int) const’
   boundary_ids.insert(1);
                        ^
```
Another id will be added after you initialize the container as `const`.

bangerth commented on examples/step-37/step-37.cc:
```diff
@@ -841,8 +841,7 @@ namespace Step37
 
-    std::set<types::boundary_id> dirichlet_boundary;
-    dirichlet_boundary.insert(0);
+    const std::set<types::boundary_id> dirichlet_boundary = {0};
```
Since you're already in this patch, would you mind terribly changing the name for the variable to `dirichlet_boundary_ids`? Because it's really a set of ids, not of boundaries.

kronbichler replied:
Not at all - I fixed this here and in step-66.

kronbichler commented:
@marcfehling I fixed this problem - and I also updated the test, because it did not actually use that function with the particular boundary id as it was supposed to be from the test description. Now it should be fine.

marcfehling commented on tests/dofs/dof_tools_09.cc:
```diff
@@ -44,7 +43,7 @@ check_this(const DoFHandler<dim> &dof_handler)
   // check for boundary id 0 and 1
   boundary_ids.insert(1);
-  DoFTools::map_dof_to_boundary_indices(dof_handler, map);
+  DoFTools::map_dof_to_boundary_indices(dof_handler, boundary_ids, map);
```
Good catch! :)

marcfehling reviewed (approved):
CI is happy. Ready to merge.
